src/seedu/addressbook/commands/Command.java

```diff
@@ -70,4 +70,11 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    /**
+     * Returns true if the Command has mutated the data
+     */
+    public boolean isMutating() {
+        throw new UnsupportedOperationException("This method should be implement in child classes");
+    }
 }
```

Comment:
Perhaps you can consider making this method abstract? For example having a default behaviour for all child classes like returning `false`, and overriding this method to return `true` whenever appropriate.